### PR TITLE
add Iterations Interval Callback

### PR DESCRIPTION
**Pull request by fabdsp**
Feature to add and execute your own callbacks after every iteration interval.

To perform clean, check etc. regularly but not at every iteration.

### CODE REVIEW

File: src/M6Web/Bundle/DaemonBundle/Command/DaemonCommand.php [hunk 1]
```diff
@@ -475,6 +484,36 @@ protected function isLastLoop()
...
+    /**
+     * Add your own callback after each iteration interval
+     *
+     * @param integer $iterationsInterval
+     * @param callable $onIterationsInterval
+     */
+    public function addIterationsIntervalCallback($iterationsInterval, callable $onIterationsInterval)
+    {
+        if (!is_integer($iterationsInterval) || ($iterationsInterval <= 0)) {
+            throw \Exception('Invalid interation interval. Must be > 0');
```
Comment (t-geindre) [inline src/M6Web/Bundle/DaemonBundle/Command/DaemonCommand.php]: \InvalidArgumentException ?

**Comment by t-geindre** [comment]
:+1:

File: README.md [hunk 2]
```diff
@@ -56,12 +56,31 @@ class MyDaemonizedCommand extends DaemonCommand
...
     {
         $output->writeln("Iteration");
 
         usleep(100000);
     }
+
+    /**
+     * executeEveryXLoops is called every 10 loops
+     */
+    protected function executeEveryXLoops(InputInterface $input, OutputInterface $output)
```
Comment (mikaelrandy) [inline README.md]: In `setup`, you call this method `executeEveryTenLoops` ;)

File: src/M6Web/Bundle/DaemonBundle/Command/DaemonCommand.php [hunk 3]
```diff
@@ -475,6 +485,36 @@ protected function isLastLoop()
...
+    /**
+     * Add your own callback after every iteration interval
+     *
+     * @param integer $iterationsInterval
+     * @param callable $onIterationsInterval
+     */
+    public function addIterationsIntervalCallback($iterationsInterval, callable $onIterationsInterval)
+    {
+        if (!is_integer($iterationsInterval) || ($iterationsInterval <= 0)) {
+            throw \InvalidArgumentException('Invalid interation interval. Must be > 0');
```
Comment (mikaelrandy) [inline src/M6Web/Bundle/DaemonBundle/Command/DaemonCommand.php]: "Must be a positive integer"

**Comment by t-geindre** [comment]
:+1:

**Comment by mikaelrandy** [comment]
:+1:

**Comment by fabdsp** [comment]
v1.4.0

**Pull request merged**